### PR TITLE
Better handle script evaluation in devtools protocol

### LINKED ISSUE #4658: devtools automationProtocol: execute/executeAsync removing 7 characters from scripts, breaking them.
**Environment (please complete the following information):**
 - **WebdriverIO version:** 5.15.1
 - **Mode:** standalone
 - **If WDIO Testrunner, running sync/async:** [e.g. sync/async]
 - **Node.js version:** 10
 - **NPM version:** yarn 1.19.1
 - **Browser name and version:** Chrome 76
 - **Platform name and version:** MacOS
 - **Additional wdio packages used (if applicable):**  devtools

**Config of WebdriverIO**

```
automationProtocol: 'devtools',
```

**Describe the bug**

```ts
const el = await this.webdriverio.$(selector);
await el.waitForExist();

// Move the "mouse" to the object.
// This is necessary for items that require hover/mouseenter before a click is recognized.
await el.scrollIntoView();
await el.moveTo(); // fails here
```

## Error

```
Unexpected token ':'
```

## Code being executed:
```
return { scrollX: this.pageXOffset, scrollY: this.pageYOffset }
```
This is from `webdriverio`:
https://github.com/webdriverio/webdriverio/blob/master/packages/webdriverio/src/utils.js#L361

It fails because `executeScript` is manipulating the string without checking to see if the change is valid:
```ts
const result = eval(script.slice(7))
```
https://github.com/webdriverio/webdriverio/blob/master/packages/devtools/src/scripts/executeScript.js#L18,

This results in attempting to run:
```ts
eval('{ scrollX: this.pageXOffset, scrollY: this.pageYOffset }');
```

This is not valid JS.

It *should* be
```ts
eval('({ scrollX: this.pageXOffset, scrollY: this.pageYOffset })');
```

(notice the parens)

**To Reproduce**

*Example 1.* see above.

*Example 2.* `await browser.execute('console.log("hello")');`

```
    Evaluation failed: SyntaxError: Unexpected token '.'
        at __puppeteer_evaluation_script__:14:30
```

*Example 3.* `await browser.execute('          return "string";');`

```
    Evaluation failed: SyntaxError: Illegal return statement
        at __puppeteer_evaluation_script__:14:30
```

*Example 4.* `await browser.execute('/* test */ console.log("test")');`

```
    Evaluation failed: SyntaxError: Unexpected token '*'
        at __puppeteer_evaluation_script__:14:30
```

*Example 5.* `executeAsync` has this issue as well.

**Expected behavior**

* `execute` and `executeAsync` should never blindly remove characters. Either require developers to provide strings already formatted correctly (and fix code [in webdriverio](https://github.com/webdriverio/webdriverio/blob/master/packages/webdriverio/src/utils.js#L361)), or carefully examine strings before making any changes.
* Ideally, any manipulation of the script should be done before sending to the chrome debug API so that the exception and logs show the actual script being executed. This was difficult to trace because the code looks fine.

**Log**

`puppeteer:protocol SEND ►`
```json
{
  "sessionId": "90ED379684DB047F703177EF7F0F78E4",
  "method": "Runtime.callFunctionOn",
  "params": {
    "functionDeclaration": "(_, script, dataProperty, dataFlag, ...args) => {\n  window.arguments = args;\n    console.log('vvvvvvvvvvvvvvvvvvvvvvvvvvvv')\n  console.log('script.slice(7)', script.slice(7));\n  try {\n      const result = eval(script.slice(7));\n\n  } catch (err) {\n\n      console.log('ERR', err);\n      console.log(script);\n      console.log('==========================')\n  }\n  const result = eval(script.slice(7));\n  console.log('result', result);\n    console.log('^^^^^^^^^^')\n  let tmpResult = result instanceof NodeList ? Array.from(result) : result;\n  const isResultArray = Array.isArray(tmpResult);\n  tmpResult = isResultArray ? tmpResult : [tmpResult];\n\n  if (tmpResult.find(r => r instanceof HTMLElement)) {\n    tmpResult = tmpResult.map((r, i) => {\n      if (r instanceof HTMLElement) {\n        const dataPropertyValue = `${dataFlag}_${i}`;\n        r.setAttribute(dataProperty, dataPropertyValue);\n        return dataPropertyValue;\n      }\n\n      return result;\n    });\n  }\n\n  return isResultArray ? tmpResult : tmpResult[0];\n}\n//# sourceURL=__puppeteer_evaluation_script__\n",
    "executionContextId": 7,
    "arguments": [
      {
        "objectId": "{\"injectedScriptId\":7,\"id\":22}"
      },
      {
        "value": "return { scrollX: this.pageXOffset, scrollY: this.pageYOffset };"
      },
      {
        "value": "data-devtoolsdriver-fetchedElement"
      },
      {
        "value": "__executeElement"
      }
    ],
    "returnByValue": true,
    "awaitPromise": true,
    "userGesture": true
  },
  "id": 107
}
```
`puppeteer:protocol ◀ RECV`
```json
{
  "method": "Runtime.consoleAPICalled",
  "params": {
    "type": "log",
    "args": [
      {
        "type": "string",
        "value": "ERR"
      },
      {
        "type": "object",
        "subtype": "error",
        "className": "SyntaxError",
        "description": "SyntaxError: Unexpected token ':'\n    at __puppeteer_evaluation_script__:6:34",
        "objectId": "{\"injectedScriptId\":7,\"id\":23}",
        "preview": {
          "type": "object",
          "subtype": "error",
          "description": "SyntaxError: Unexpected token ':'\n    at __puppeteer_evaluation_script__:6:34",
          "overflow": false,
          "properties": [
            {
              "name": "stack",
              "type": "string",
              "value": "SyntaxError: Unexpected token ':'\n    at __puppeteer_evaluation_script__:6:34"
            },
            {
              "name": "message",
              "type": "string",
              "value": "Unexpected token ':'"
            }
          ]
        }
      }
    ],
    "executionContextId": 7,
    "timestamp": 1571419795871.338,
    "stackTrace": {
      "callFrames": [
        {
          "functionName": "",
          "scriptId": "22",
          "url": "__puppeteer_evaluation_script__",
          "lineNumber": 9,
          "columnNumber": 14
        }
      ]
    }
  },
  "sessionId": "90ED379684DB047F703177EF7F0F78E4"
} 
```

**Additional context**

We need to put all javascript that should execute in browser in a string so that TypeScript doesn't expect it to be typed and our code coverage tooling doesn't attempt to decorate it or count it towards coverage.

### PR DESCRIPTION
**Pull request by christian-bromann**
## Proposed changes

closes #4658

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/devtools/src/commands/executeAsyncScript.js [hunk 1]
```diff
@@ -5,6 +5,16 @@ import { SERIALIZE_PROPERTY, SERIALIZE_FLAG } from '../constants'
 export default async function executeAsyncScript ({ script, args }) {
     const page = this.getPageHandle(true)
     const scriptTimeout = this.timeouts.get('script')
+    script = script.trim()
+
+    if (script.startsWith('return (')) {
```
Comment (mgrybyk) [inline packages/devtools/src/commands/executeAsyncScript.js]: Do we have test for this one?

Comment (christian-bromann) [reply packages/devtools/src/commands/executeAsyncScript.js]: yes, see in `e2e/protocol.test.js`
```js
    it('can handle various of script formats', async () => {
        expect(await browser.executeScript('console.log("hello")')).toBe(undefined)
        expect(await browser.executeScript(' return "string";')).toBe('string')
        expect(await browser.executeScript('/* test */ console.log("test")')).toBe(undefined)
        expect(await browser.executeScript('return { foo: "bar" }')).toEqual({ foo: 'bar' })
    })
```

Comment (christian-bromann) [reply packages/devtools/src/commands/executeAsyncScript.js]: we test all commands currently like that as it just makes more sense to have one common e2e test.

Comment (mgrybyk) [reply packages/devtools/src/commands/executeAsyncScript.js]: Should we have there script that starts with `return (`?

Comment (christian-bromann) [reply packages/devtools/src/commands/executeAsyncScript.js]: added such a test

Comment (mgrybyk) [reply packages/devtools/src/commands/executeAsyncScript.js]: thanks

**Pull request merged**